doc/Language/rb-nutshell.pod

```diff
@@ -0,0 +1,944 @@
+
+Classes are defined similarly between Ruby and Perl 6, using the C<class>
+keyword. Ruby uses C<def> for methods, whereas Perl 6 uses C<method>.
+
+    # Ruby
+    class Foo
+        def greet(name)
+            puts "Hi #{name}!"
+        end
+    end
+
+    # Perl 6
+    class Foo {
+        method greet($name) {
+            say "Hi $name!"
+        }
+    }
+
+In Ruby you can use an attribute without declaring it beforehand, and you can
+tell it is an attribute because of the C<@> sigil. In Perl 6 you must use a
+C<has> declaration and a variety of sigils. You can use the C<!> twigil for
+private attributes or C<.> to create an accessor. 
+
+    # Ruby
+    class Person
+        attr_accessor :age
+        def initialize
+            @name = 'default'
+        end
+    end
+
+    # Perl 6
+    class Person
+        has $!name = 'default';
+        has $.age;
+    end
+
+Creating a new instance of the class uses the C<.new> method. In Perl 6 you get
+a default constructor that accepts key/value pairs of accessor attributes, and
+can do further setup in the C<BUILD> method.
+
+    # Ruby
+    class Person
+        attr_accessor :name, :age
+        def initialize(attrs)
+            @name = attrs[:name] || 'Jill'
+            @age  = attrs[:age] || 42
+            @birth_year = Time.now.year - @age
+        end
+    end
+    p = Person.new( name: 'Jack', age: 23 )
+
+    # Perl 6
+    class Person
+        has $.name = 'Jill';
+        has $.age  = 42;
+        has $!birth_year;
+        method BUILD {
+            $!birth_year = now.Date.year - $.age;
+        }
+    }
+    p = Person.new( name => 'Jack', age => 23 )
+
+=head3 Private Methods
+
+Private methods in Perl 6 are declared with a C<!> prefixed in their name, and
+are invoked with a C<!> instead of a C<.>.
+
+    # Ruby
+    class Foo
+        def visible
+            puts "I can be seen!"
+            hidden
+        end
+
+        private
+        def hidden
+            puts "I cannot easily be called!"
+        end
+    end
+
+    # Perl 6
+    class Foo {
+        method visible {
+            say "I can be seen!"
+            self!hidden
+        }
+
+        method !hidden {
+            say "I cannot easily be called!"
+        }
+    }
+
+An important note is that in Ruby child objects can see parent private methods
+(so they are more like "protected" methods). In Perl 6 child objects cannot
+call parent private methods.
+
+=head3 Going Meta
+
+Here are a few examples of meta-programming. Note that Perl 6 separates the
+meta-methods from the regular methods.
+
+    person.class # Ruby, returns Person
+    person.WHAT  # Perl 6, returns Person
+
+    person.methods          # Ruby
+    $person.^methods        # Perl 6. The .^ syntax is sugar
+    $person.HOW.methods($p) # Perl 6, not using the .^ sugar
+
+    person.instance_variables # Ruby
+    person.^attributes        # Perl 6
+
+In Perl 6, everything is an object, but not all operations are equivalent to
+C<.send>.
```

Comment:
Elaborate that `+` as an example is a multi-sub that dispatches based on the types, as opposed to a method ON those types.